### PR TITLE
Proto encode functions for AccessPathProto, ClaimProto and CheckProto

### PR DESCRIPTION
**Pull request by csilvestrini**

### CODE REVIEW

File: java/arcs/core/data/proto/AccessPathProtoDecoder.kt [hunk 1]
```diff
@@ -22,13 +22,41 @@ fun AccessPathProto.decode(
...
+            val connectionRoot = root as AccessPath.Root.HandleConnection
+            proto.particleSpec = connectionRoot.particle.spec.name
+            proto.handleConnection = connectionRoot.connectionSpec.name
+        }
+        is AccessPath.Root.HandleConnectionSpec -> {
+            val connectionSpecRoot = root as AccessPath.Root.HandleConnectionSpec
+            proto.particleSpec = connectionSpecRoot.particleSpecName
+            proto.handleConnection = connectionSpecRoot.connectionSpec.name
+        }
+        else -> throw UnsupportedOperationException("Unsupported AccessPath.Root type: $this")
```
Comment (bgogul) [inline java/arcs/core/data/proto/AccessPathProtoDecoder.kt:47]: There is also AccessPath.Root.Store, but I guess we cannot encode that yet as it is not in the proto? I filed b/161947321. Can you reference it here?

File: java/arcs/core/data/proto/AccessPathProtoDecoder.kt [hunk 2]
```diff
@@ -22,13 +22,41 @@ fun AccessPathProto.decode(
...
+fun AccessPath.encode(): AccessPathProto {
+    val proto = AccessPathProto.newBuilder()
+        .addAllSelectors(selectors.map { it.encode() })
+    when (root) {
+        is AccessPath.Root.Handle -> {
+            val handleRoot = root as AccessPath.Root.Handle
+            proto.handleConnection = handleRoot.handle.name
         }
+        is AccessPath.Root.HandleConnection -> {
+            val connectionRoot = root as AccessPath.Root.HandleConnection
```
Comment (bgogul) [inline java/arcs/core/data/proto/AccessPathProtoDecoder.kt:38]: This is unfortunate as smart casts are not supported across packages.

File: java/arcs/core/data/proto/AccessPathProtoDecoder.kt (hunk 1 above)
Comment (csilvestrini) [reply java/arcs/core/data/proto/AccessPathProtoDecoder.kt:47]: Done

File: java/arcs/core/data/proto/AccessPathProtoDecoder.kt (hunk 2 above)
Comment (csilvestrini) [reply java/arcs/core/data/proto/AccessPathProtoDecoder.kt:38]: Yeah, it's annoying.. I never really understood why that doesn't work.

**Pull request merged**